### ISSUE TITLE
ROM Doesn't Generate Correct Verilog

### ISSUE THREAD

**Issue by ben-k** [ucb-bar/chisel#155]
I am generating a ROM in Chisel like so:

  val angles = (0 until params.l).map(t => UInt(round((atan(pow(2.0,-1.0*t)).toDegrees / 360) \* pow(2,params.l)),   width = params.l) )  // Just a bunch of math to generate an array
    val atanTable = Vec( angles )

I then access the ROM via a counter register called iter by calling atanTable(iter). So a case statement in Verilog is being built like this:

  assign T7 =
      iter == 5'd0 ? 16'h2000
    : iter == 5'd1 ? 16'h12e4
      ... // 13 cases removed for brevity
    : iter == 5'd15 ? 16'h0
`ifndef SYNTHESIS
    :$random()
`endif
    ;

This breaks the DC Verilog parser.  I thought the ifndef statement was the problem, but even if you remove that, it still doesn't parse because the last conditional is short a clause.  Something like this would be correct syntax:

  assign T7 =
      iter == 5'd0 ? 16'h2000
    : iter == 5'd1 ? 16'h12e4
      ... // 13 cases removed for brevity
    : iter == 5'd15 ? 16'h0: 16'h0;

In short, I'm not sure what the intended behavior of the Chisel generator is here, but it definitely isn't producing valid Verilog.
